# HG changeset patch
# User Gary Kramlich <grim@reaperworld.com>
# Date 1570433439 18000
#      Mon Oct 07 02:30:39 2019 -0500
# Branch release-2.x.y
# Node ID e3c4fde8a94a25bb571dc7932bdd6cca086f8ae8
# Parent  a8e08676e356c7d003852fddd3d9c3826dcfa1d6
G_CONST_RETURN has been deprecated for awhile and honestly it's not necessary for us

diff -r a8e08676e356 -r e3c4fde8a94a libpurple/util.c
--- a/libpurple/util.c	Fri Aug 02 23:56:19 2019 +0000
+++ b/libpurple/util.c	Mon Oct 07 02:30:39 2019 -0500
@@ -4695,7 +4695,7 @@
  * This function is copied from g_strerror() but changed to use
  * gai_strerror().
  */
-G_CONST_RETURN gchar *
+const gchar *
 purple_gai_strerror(gint errnum)
 {
 #if GLIB_CHECK_VERSION(2, 32, 0)
diff -r a8e08676e356 -r e3c4fde8a94a libpurple/util.h
--- a/libpurple/util.h	Fri Aug 02 23:56:19 2019 +0000
+++ b/libpurple/util.h	Mon Oct 07 02:30:39 2019 -0500
@@ -1394,7 +1394,7 @@
  * @return The UTF-8 error message.
  * @since 2.4.0
  */
-G_CONST_RETURN gchar *purple_gai_strerror(gint errnum);
+const gchar *purple_gai_strerror(gint errnum);
 
 /**
  * Compares two UTF-8 strings case-insensitively.  This comparison is
# HG changeset patch
# User Gary Kramlich <grim@reaperworld.com>
# Date 1570433500 18000
#      Mon Oct 07 02:31:40 2019 -0500
# Branch release-2.x.y
# Node ID 9d4485060df8bde34d576346f8e22663bc670fb9
# Parent  e3c4fde8a94a25bb571dc7932bdd6cca086f8ae8
Disable deprecation warnings for the functions that have GParameter in their arguments

diff -r e3c4fde8a94a -r 9d4485060df8 libpurple/media.h
--- a/libpurple/media.h	Mon Oct 07 02:30:39 2019 -0500
+++ b/libpurple/media.h	Mon Oct 07 02:31:40 2019 -0500
@@ -161,8 +161,10 @@
  *
  * @since 2.8.0
  */
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 void purple_media_set_params(PurpleMedia *media,
 		guint num_params, GParameter *params);
+G_GNUC_END_IGNORE_DEPRECATIONS
 
 /**
  * Gets the list of optional parameters supported by the media backend.
@@ -208,10 +210,12 @@
  *
  * @since 2.6.0
  */
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 gboolean purple_media_add_stream(PurpleMedia *media, const gchar *sess_id,
 		const gchar *who, PurpleMediaSessionType type,
 		gboolean initiator, const gchar *transmitter,
 		guint num_params, GParameter *params);
+G_GNUC_END_IGNORE_DEPRECATIONS
 
 /**
  * Gets the session type from a session
